docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+    df["similarities"] = df.embeddings.apply(lambda x: np.dot(x, query_embedding))
+    most_similar = df.sort_values("similarities", ascending=False).iloc[0]["texts"]
+
+    prompt = SoccerPrompt(context=most_similar, question=query)
+    completion = chat.create(prompt)
+    print(completion)
+
+
+countries = ["English", "Spanish", "German", "Italian"]
+for country in countries:
+    ask_soccer(f"Who won the {country} top flight?")
+```
+
+## A More Realistic Approach...
+
+!!! note
+
+    This tutorial's code snippets have been rearranged to clearly demonstrate each step of the functionality. For a runnable script, take a look at the [code](https://github.com/Mirascope/mirascope/blob/main/cookbook/basic_rag_example/pinecone_embeddings.py) in our repo. Note that the repo code contains additional functionality and logic flows to make re-runs of the script as efficient as possible.
+
+In reality, we could have passed the entirety of `soccer_info` into the prompt due to its small size. Let's take a look at a practical, scaled up example using the vector database Pinecone to manage the larger data.
+
+Our dataset will be a collection of news articles from 2004 pulled from this [public Github database.](https://raw.githubusercontent.com/Dawit-1621/BBC-News-Classification/main/Data/BBC%20News%20Test.csv) We will pick a topic, then ask our chat model to give a summary of 4 relevant articles.
+
+### Dataset/Preprocessing
+
+Chat and embedding models have token limits on a single call - this wasn't an issue in the soccer example, but these constraints come into play with much larger texts and datasets. For the purposes of this recipe, we split the large texts into roughly equal sizes using helper function `split_text()`. Since we want to be able to fit multiple articles into our final prompt, let's set `max_tokens=1000` and split the text into how many thousands of tokens the original text comprises of.
+
+```python
+def split_text(text: str, tokens: list[int], max_tokens: int) -> list[str]:
+    """Roughly splits a text into chunks according to max_tokens.
+
+    Text is split into equal word counts, with number of splits determined by how many
+    times `max_tokens` goes into the total number of tokens (including partially). Note
+    that tokens and characters do not have an exact correspondence, so in certain edge
+    cases a chunk may be slightly larger than max_tokens.
+
+    Args:
+        text: The text to split.
+        tokens: How many tokens `text` is.
+        max_tokens: The (rough) number of maximum tokens per chunk.
+
+    Returns:
+        A list of the split texts.
+    """
+    words = text.split()
+    num_splits = len(tokens) // max_tokens + 1
+    split_texts = []
+
+    for i in range(num_splits):
+        start = i * len(words) // num_splits
+        end = (i + 1) * len(words) // num_splits
+        split_texts.append(" ".join(words[start:end]))
+
+    return split_texts
+```
+
+```python
```

Comment:
instead of having two chunks of code back-to-back, you should split these up with text explaining each code chunk itself.